docs/semgrep-secrets/conceptual-overview.md

```diff
@@ -0,0 +1,254 @@
+* (For GitHub repositories) If these secrets are in public or private repositories.
+
+Save time and effort by prioritizing valid leaked secrets. Inform developers of valid, active secrets in their GitHub PRs by posting PR comments. 
+
+This document explains how Semgrep Secrets works and its approach to detecting secrets.
+
+:::info
+For a guide to setting up Semgrep Secrets, see [<i class="fa-regular fa-file-lines"></i> Getting started with Semgrep Secrets](/semgrep-secrets/getting-started).
+:::
+
+:::tip
+Secrets scanning is **language-agnostic**. This means Semgrep Secrets can scan any language for secrets.
+:::
+
+To ensure that findings are high-signal, comprehensive, and easy for users to prioritize, a secrets scan performs the following:
+
+* Semantic analysis
+* Validation
+* Entropy analysis
+
+The following sections explain how each analysis works.
+
+## Detecting secrets through semantic analysis
+
+Semantic analysis refers to Semgrep's ability to understand how data is used within your code. This differentiates Semgrep from regex-based detectors that simply define a pattern to match to a piece of code.
+
+Semgrep uses several mechanisms to perform semantic analysis. In particular, Semgrep uses **data-flow analysis**, which means that it is able to track data, such as variables, and the flow of that data across files and functions in your codebase. Semgrep is able to detect if a variable is renamed, unsanitized or sanitized, reassigned, or used in a function in such a way that a secret is exposed.
+
+See the following Semgrep rule and JavaScript test code for an example.
+
+<iframe title="AWS hardcoded access key" src="https://semgrep.dev/embed/editor?snippet=EPj5" width="100%" height="432px" frameBorder="0"></iframe>
+<br />
+
+ The Semgrep rule detects hardcoded AWS secret access keys. The test code defines an access key in the variable `secret`. Click **<i class="fa-solid fa-play"></i> Run** to see the true positives.
+
+* **Line 2:** Semgrep and regex-based scanners can detect both the variable name `secret` and its value (token) in line 2.
+	* A regex-based scanner may generate a noisy finding from line 2 even though `secret` has not been passed to any function.
+	* Semgrep doesn't generate a finding because the token hasn't been passed as a `secretAccessKey` or similar.
+* **Line 7:** Semgrep and regex-based scanners are able to detect **line 6**, in which the plain-text secret is passed to the `AWS.config.update` function.
+* **Line 17:** Semgrep and regex-based scanners can detect **line 14**, in which `secret` is passed.
+* **Line 26:** Semgrep correctly skips `conf.secret` in **line 21**. Regex-based scanners simply looking for matches of the string `secret` generate a false positive.
+
+## Validating secrets 
+
+After scanning for secrets, Semgrep uses a **post-processor** function called a **validator** to determine if a secret is actively being used, or some other state.
```

Comment:
"After scanning for **secrets**, Semgrep **Secrets**...."

While I was writing, I was wondering if it's possible to use Semgrep interchangeably with Semgrep Secrets. Though from a TW standpoint, the convention for consistency is to always spell out the product. Maybe I can use "credentials" occasionally as a common-noun synonym in place of secrets, so that the redundancy of "secrets" isn't so severe? So it'd be:

"After scanning for credentials, Semgrep Secrets..."

Reply:
Yeah that could work, if sprinkled lightly. Alternatively: “After scanning, Semgrep Secrets uses…”